gc/base/standard/Scavenger.hpp

```diff
@@ -263,7 +263,32 @@ class MM_Scavenger : public MM_Collector
 	 * @return Whether or not objectPtr should be remembered.
 	 */
 	MMINLINE bool scavengeObjectSlots(MM_EnvironmentStandard *env, MM_CopyScanCacheStandard *scanCache, omrobjectptr_t objectPtr, uintptr_t flags, omrobjectptr_t *rememberedSetSlot);
-	MMINLINE MM_CopyScanCacheStandard *incrementalScavengeObjectSlots(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr, MM_CopyScanCacheStandard* scanCache);
+	MMINLINE MM_CopyScanCacheStandard *incrementalScavengeObjectSlots(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr, MM_CopyScanCacheStandard* scanCache);	
+	
+	/**
+	 * For fast traversal of deep structure nodes - scan objects with self referencing fields with priority
+	 * Split into two functions deepScan and deepScanOutline. Frequently called checks (see lazy start check) must be inlined
+	 * @param env The environment.
+	 * @param objectPtr The pointer to the object.
+	 * @param selfReferencingField1 Offset to the first priority field of the object
+	 * @param selfReferencingField2 Offset to the second priority field, if it can't follow through in one direction, 
+	 * it will attempt to use the second self referencing field 
+	 */
+	MMINLINE void
+	deepScan(MM_EnvironmentStandard *env, omrobjectptr_t objectPtr, uintptr_t selfReferencingField1, uintptr_t selfReferencingField2)
+	{
+		
+	   /**
+		* Lazy Start condition used for gatekeeping - inhibit the special treatment routine with relatively high probability 
+		* to skip over most false positives (shorter lists), while only marginally delay detection of very deep structures.
+		* Check last few LSB of the object address for probability 1/16
+		*/	
+		if(0 == ((uintptr_t)objectPtr & 0x78)) {
```

Comment:
space after `if`

Reply:
You could put this condition into another inlined helper, perhaps name it shouldStartDeepScan.
Then split the comment. First part, what it does stays here. Last part, how it does,  goes into the helper.